lib/puma/cluster.rb

```diff
@@ -488,7 +518,7 @@ def run
                 w.pid = pid if w.pid.nil?
               end
 
-              if w = @workers.find { |x| x.pid == pid }
+              if w = @mold&.pid == pid ? @mold : @workers.find { |x| x.pid == pid }
```

Comment:
molds still use the same stat_thread logic so make sure it's handled on any of these signals